app/src/main/java/com/keylesspalace/tusky/components/account/AccountActivity.kt

```diff
@@ -491,6 +497,33 @@ class AccountActivity : BottomSheetActivity(), ActionButtonActivity, MenuProvide
         }
     }
 
+    private fun updateBadges() {
+        binding.accountBadgeContainer.removeAllViews()
+
+        val isLight = resources.getBoolean(R.bool.lightNavigationBar)
+
+        if (loadedAccount?.bot == true) {
+            val badgeView = getBadge(getColor(R.color.tusky_grey_50), R.drawable.ic_bot_24dp, getString(R.string.profile_badge_bot_text), isLight)
+            binding.accountBadgeContainer.addView(badgeView)
+        }
+
+        loadedAccount?.roles?.forEach { role ->
+            val badgeColor = if (role.color.isEmpty()) {
+                // sometimes the color is not set for a role, in this case fall back to our default blue
+                getColor(R.color.tusky_blue)
+            } else {
+                Color.parseColor(role.color)
+            }
+
+            val sb = SpannableStringBuilder("${role.name} ${viewModel.domain}")
```

Comment:
Out of curiosity: Is translation an issue here? This does only depend on the language of the server?

Reply:
The role has only one name. Admins can input whatever they want. So they probably use a name that is meaningful in the servers main language. It can't be translated.

Reply:
I think Mastodon comes with some default roles ("Owner", "Admin", "Moderator") to map the old permission system to, but admins can change them however they like.